site/tests/app/controllers/HomePageControllerTester.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace tests\app\controllers;
+
+use app\libraries\response\RedirectResponse;
+use app\libraries\response\WebResponse;
+use app\models\User;
+use app\libraries\Core;
+use tests\BaseUnitTest;
+use app\controllers\HomePageController;
+
+class HomePageControllerTester extends BaseUnitTest {
+
+	protected $controller;
+
+	protected $core;
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->core = $this->createMockCore([
+			'getUsernameChangeText' => True
+		], [
+			'accessGrading' => True
+		], [
+			'updateUser' => []
+		], [], [
+			'addSuccessMessage' => true,
+			'addErrorMessage' => true,
+			'buildUrl' => 'http://192.168.56.111/home'
+		]);
+
+		$this->true_controller = new HomePageController($this->core);
+		$this->controller = $this->getMockBuilder(HomePageController::class)
+			->setConstructorArgs([$this->core])
+			->setMethods(['changePassword', 'getCourses'])
+			->getMock();
+		$this->controller->method('changePassword')
+			->will($this->returnCallback(function ($post) {
+				$this->mocked_methods['changePassword'] = true;
+				$_POST = $post;
+				return $this->true_controller->changePassword();
+			}));
+		$courses = (object) [
+			"json_response" => (object) [
+				"json" => $this->getCoursesResponse()
+			]
+		];
+		$this->controller->method('getCourses')
+			->willReturn($courses);
+	}
+
+	public function testHomePageViewChangePasswordSuccess() {
+		$_POST = [
```

Comment:
I don't think you should be using the POST super global variable here. Just create a regular variable.